pkg/server/api/trust/v1/resolver.go

```diff
@@ -0,0 +1,287 @@
+	return UserFromContext(ctx)
+}
+
+// TokenAccessFromContext implements auth.ContextAccessor interface
+func (r *Resolver) TokenAccessFromContext(ctx context.Context) *auth.TokenAccessData {
+	return TokenAccessFromContext(ctx)
+}
+
+func NewMux(
+	logger *log.Logger,
+	usrmgrSvc *usrmgr.Service,
+	trustSvc *trust.Service,
+	authCfg AuthConfig,
+) *chi.Mux {
+	r := chi.NewMux()
+
+	r.Handle("/graphql", graphqlHandler(logger, usrmgrSvc, trustSvc, authCfg))
+
+	r.Handle("/playground", playground.Handler("GraphQL Playground", "/api/trust/v1/graphql"))
+
+	r.Post("/auth/authenticate", authTokenHandler(trustSvc, authCfg))
+	r.Delete("/auth/logout", trustCenterLogoutHandler(authCfg))
+
+	return r
+}
+
+func graphqlHandler(logger *log.Logger, usrmgrSvc *usrmgr.Service, trustSvc *trust.Service, authCfg AuthConfig) http.HandlerFunc {
+	var mb int64 = 1 << 20
+
+	resolver := &Resolver{
+		trustCenterSvc: trustSvc,
+		authCfg:        authCfg,
+	}
+
+	c := schema.Config{
+		Resolvers: resolver,
+	}
+
+	c.Directives.MustBeAuthenticated = auth.MustBeAuthenticatedDirective(resolver)
+
+	es := schema.NewExecutableSchema(c)
+
+	srv := handler.New(es)
+
+	srv.AddTransport(transport.POST{})
+	srv.AddTransport(transport.GET{})
+	srv.AddTransport(transport.Options{})
+	srv.AddTransport(
+		transport.MultipartForm{
+			MaxMemory:     32 * mb,
+			MaxUploadSize: 50 * mb,
+		},
+	)
+
+	srv.Use(extension.Introspection{})
+
+	srv.SetRecoverFunc(func(ctx context.Context, err any) error {
+		logger := httpserver.LoggerFromContext(ctx)
+		logger.Error("resolver panic", log.Any("error", err), log.Any("stack", string(debug.Stack())))
+
+		return errors.New("internal server error")
+	})
+
+	return WithSession(usrmgrSvc, trustSvc, authCfg, srv.ServeHTTP)
+}
+
+// TrustService returns a trust service scoped to the given tenant
+func (r *Resolver) TrustService(ctx context.Context, tenantID gid.TenantID) *trust.TenantService {
+	return r.trustCenterSvc.WithTenant(tenantID)
+}
+
+// GetTenantService returns a tenant service for the given tenant ID
+func (r *Resolver) GetTenantService(ctx context.Context, tenantID gid.TenantID) *trust.TenantService {
+	return r.trustCenterSvc.WithTenant(tenantID)
+}
+
+func WithSession(usrmgrSvc *usrmgr.Service, trustSvc *trust.Service, authCfg AuthConfig, next http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		if authCtx := trySessionAuth(ctx, w, r, usrmgrSvc, authCfg); authCtx != nil {
+			next(w, r.WithContext(authCtx))
+			updateSessionIfNeeded(authCtx, usrmgrSvc)
+			return
+		}
+
+		if authCtx := tryTokenAuth(ctx, w, r, trustSvc, authCfg); authCtx != nil {
+			next(w, r.WithContext(authCtx))
+			return
+		}
+
+		next(w, r.WithContext(ctx))
+	}
+}
+
+func trySessionAuth(ctx context.Context, w http.ResponseWriter, r *http.Request, usrmgrSvc *usrmgr.Service, authCfg AuthConfig) context.Context {
+	cookieValue, err := securecookie.Get(r, securecookie.DefaultConfig(
```

Comment:
cookie coinfivguration must be exposed over a package and use in console v1 and trust v1